### ISSUE TITLE
Render quick input separators as individual items instead of applying border to list item

### ISSUE THREAD

**Issue by daviddossett** [microsoft/vscode#169296]
Here's what the borders look like today. Note the gap on both sides that results from L/R padding used to get the spacing effect on list items.

<img width="658" alt="CleanShot 2022-12-15 at 09 44 09@2x" src="https://user-images.githubusercontent.com/25163139/207931312-57aab45d-e93b-4d72-864d-04c9101fbce2.png">

Because separators are rendered on the list items themselves, there are two problems:
- I can't render them to take up the full width of the quick pick as desired (see context menu example below)
- I can't change their height to better separate sections within a quick pick list

## Prior art

<img width="324" alt="CleanShot 2022-12-15 at 09 44 24@2x" src="https://user-images.githubusercontent.com/25163139/207931804-dd666fa7-daf6-4000-b99a-8bebaa845f0d.png">
<img width="314" alt="CleanShot 2022-12-15 at 09 45 28@2x" src="https://user-images.githubusercontent.com/25163139/207931807-83137a9e-8df3-4f42-8c27-7bdba0bdf0f1.png">

## Example

<img width="968" alt="CleanShot 2022-12-15 at 09 51 48@2x" src="https://user-images.githubusercontent.com/25163139/207932442-c1c20c0a-31c4-4621-a3f6-34ffd9791a3d.png">

<img width="987" alt="CleanShot 2022-12-15 at 09 51 54@2x" src="https://user-images.githubusercontent.com/25163139/207932457-de01c058-f865-4df8-91e0-5652bef15215.png">

**Comment by TylerLeonhardt** [comment]
Makes sense. Might help consolidate the code I wrote to allow separators to have buttons.
